eulerpi/core/result_manager.py

```diff
@@ -428,37 +431,43 @@ def load_slice_inference_results(
                 ndmin=2,
             )
             if i == 0:
-                overall_density_evals = density_evals
-                overall_sim_results = sim_results
-                overall_params = params
+                param_dim = params.shape[1]
+                data_dim = sim_results.shape[1]
+                overall_density_evals = density_evals.reshape(
```

Comment:
What will -1 resolve to? If we already know the dimension, I think we should use the dimension.
You could also get the number of steps here, if we know the data / param / density dimension.